### PR TITLE
Config

### PR DESCRIPTION
**Pull request by grantleehoffman**
* Add config section to project files
* Add plugin_config section to project files
* Add `!config` options in scopes with mapping to config and plugin_configs
* Add abilitity to pass `!env` & `!include` tag values to project files
* Update docs to include config information

### CODE REVIEW

File: nodestream/pipeline/pipeline_file_loader.py [hunk 1]
```diff
@@ -4,6 +4,8 @@
 
 from yaml import SafeLoader, load
 
+from nodestream.pipeline.scope_config import ScopeConfig
```
Comment (zprobst) [inline nodestream/pipeline/pipeline_file_loader.py:7]: this is a nitpick, but the rest of the imports are relative. Can we change this one to be?

File: nodestream/pipeline/pipeline_file_loader.py [hunk 2]
```diff
@@ -37,12 +47,27 @@ def configure(cls):
...
     @classmethod
-    def load_file_by_path(cls, file_path: str):
-        PipelineFileSafeLoader.configure()
+    def load_file_by_path(cls, file_path: str, config: ScopeConfig = None):
+        PipelineFileSafeLoader.configure(config)
         with open(file_path) as fp:
             return load(fp, cls)
 
 
+class NodestreamProjectFileSafeLoader(SafeLoader):
```
Comment (zprobst) [inline nodestream/pipeline/pipeline_file_loader.py:56]: Should we move this to the `project` package? Its nice to have the file loaders in the same place. What do you think?

File: nodestream/pipeline/scope_config.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+        Args:
+            data (Dict): The file data to create the config from.
+
+        Returns:
+            Config: The config created from the file data.
+        """
+
+        config = cls(data)
+        return config
```
Comment (zprobst) [inline nodestream/pipeline/scope_config.py:36]: Just a silly nitpick. 

```suggestion
        return cls(data)
```

File: nodestream/pipeline/scope_config.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
+from typing import Dict
+
+from ..file_io import LoadsFromYamlFile
+
+
+class ScopeConfig(LoadsFromYamlFile):
```
Comment (zprobst) [inline nodestream/pipeline/scope_config.py:6]: We can remove the implementations of `__init__` and and `__eq__` if we make this a [dataclass](https://docs.python.org/3/library/dataclasses.html).

File: nodestream/project/pipeline_scope.py [hunk 5]
```diff
@@ -1,6 +1,8 @@
 from importlib import resources
 from typing import Dict, Iterable, List
 
+from nodestream.pipeline.scope_config import ScopeConfig
```
Comment (zprobst) [inline nodestream/project/pipeline_scope.py:4]: Another nitpick for the relative import.

File: nodestream/project/run_request.py [hunk 6]
```diff
@@ -1,5 +1,7 @@
 from dataclasses import dataclass
 
+from nodestream.pipeline.scope_config import ScopeConfig
```
Comment (zprobst) [inline nodestream/project/run_request.py:3]: the same relative import nit

File: docs/docs/guides/project-plugins.md [hunk 7]
```diff
@@ -95,3 +95,31 @@ Depending on how you are building your package, you can register your project pl
...
+=== "plugin-pipeline.yaml"
+    ```
+    - implementation: myPlugin.testPipeline:TestExctractor
+      arguments:
+        base_url: !config 'base_url'
+        username: !config 'username'
+        password: !config 'password'
+    ```
+
+End users can provide values for the `!config` plugin tags in their nodestream.yaml file. This feature is particularly useful for supplying user-provided information such as URLs and credentials. The values can be accessed at plugin_config.<plugin_name>.<config_value>. To ensure proper provision of configuration values, the plugin name under The plugin name under `plugin_config` must match the plugin scope name. This enables plugins to use similar configuration value keys without conflicts.
```
Comment (ccloes) [inline docs/docs/guides/project-plugins.md:113]:
```suggestion
End users can provide values for the `!config` plugin tags in their nodestream.yaml file. This feature is particularly useful for supplying user-provided information such as URLs and credentials. The values can be accessed at plugin_config.<plugin_name>.<config_value>. To ensure proper provision of configuration values, the plugin name under `plugin_config` must match the plugin scope name. This enables plugins to use similar configuration value keys without conflicts.
```

File: nodestream/pipeline/pipeline_file_loader.py (hunk 1 above)
Comment (grantleehoffman) [reply nodestream/pipeline/pipeline_file_loader.py:7]: Sure, will do!

File: docs/docs/guides/project-plugins.md [hunk 8]
```diff
@@ -95,3 +95,31 @@ Depending on how you are building your package, you can register your project pl
...
+### Configuration for Project Plugins
+
+The `!config` YAML tag followed by a key can be used to provide end-user configurable values for project plugin pipelines.
+
+#### Plugin Creator Provided
+=== "plugin-pipeline.yaml"
+    ```
+    - implementation: myPlugin.testPipeline:TestExctractor
+      arguments:
+        base_url: !config 'base_url'
```
Comment (ccloes) [inline docs/docs/guides/project-plugins.md:108]: Should we change the name of the var to not match the argument so it is explicit about them being different between argument and config value?
```suggestion
        base_url: !config 'config_base_url'
```

then in plugin_config, reference "config_base_url"

**Review by ccloes — APPROVED** [review]
Couple comments on the documentation clarity.

File: docs/docs/guides/project-plugins.md (hunk 8 above)
Comment (grantleehoffman) [reply docs/docs/guides/project-plugins.md:108]: yeah, i'll update

**Review by zprobst — DISMISSED** [review]

**Review by zprobst — APPROVED** [review]

**Pull request merged**